baseplate/server/__init__.py

```diff
@@ -208,11 +216,29 @@ def load_and_run_script():
         help="path to a configuration file")
     parser.add_argument("entrypoint", type=_load_factory,
         help="function to call, e.g. module.path:fn_name")
+    parser.add_argument('args', nargs=argparse.REMAINDER,
+        help="arguments to pass along to the invoked script")
 
     args = parser.parse_args(sys.argv[1:])
     config = read_config(args.config_file, server_name=None, app_name=args.app_name)
     configure_logging(config, args.debug)
-    args.entrypoint(config.app)
+
+    _check_fn_signature(args.entrypoint, args.args)
+
+    args.entrypoint(config.app, args.args)
+
+
+def _check_fn_signature(script_fn, fn_args):
+    argspec = getargspec(script_fn)
+    func_sig_arg_count = len(argspec.args[1:])  # slice off app_config arg
+    cli_arg_count = len(fn_args)
```

Comment:
if we're already looking at function signatures, would it make sense to just do two arguments rather than the `*args` thing? then we only need to handle 1 or 2 args on the function signature

Reply:
I think you're saying instead of defining the script function as:
```
def run(app_config: AppConfig, *args: str):
    pass
```

You'd prefer this?
```
def run(app_config: AppConfig, args: List[str]):
    pass
```

Reply:
yessir

Reply:
(since that leaves us better open to future arguments should we need them, and is a bit easier to work with as an application)

Reply:
I'm working on this, but I'm starting to think that this might actually end up being more complicated code. We have to have a check to determine if the script function will accept the second arg and conditionally pass it in if we want to be backwards compatible.